### PR TITLE
fix: add documentation to the fields in Metadata

### PR DESCRIPTION
**Pull request by sitaram-kalluri**
**- What I did**
Add dart docs around the fields in the Metadata

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: at_commons/lib/src/keystore/at_key.dart [hunk 1]
```diff
@@ -269,25 +276,79 @@ class PrivateKey extends AtKey {
 }
 
 class Metadata {
+  /// Represents the time beyond which the key expires. Accepts time in milliseconds
```
Comment (murali-shris) [inline at_commons/lib/src/keystore/at_key.dart]: Represents the time in milliseconds....
we can remove the second sentence

**Review by murali-shris — APPROVED** [review]

File: at_commons/lib/src/keystore/at_key.dart [hunk 2]
```diff
@@ -269,25 +276,79 @@ class PrivateKey extends AtKey {
 }
 
 class Metadata {
+  /// Represents the time in milliseconds beyond which the key expires
   int? ttl;
+
+  /// Represents the time in milliseconds from when the key becomes active
```
Comment (gkc) [inline at_commons/lib/src/keystore/at_key.dart:282]: This needs to be corrected. ttb is the time UNTIL the key becomes active.

File: at_commons/lib/src/keystore/at_key.dart [hunk 3]
```diff
@@ -269,25 +276,79 @@ class PrivateKey extends AtKey {
...
   int? ttb;
+
+  /// Represents the time frequency in seconds when the cached key gets refreshed
```
Comment (gkc) [inline at_commons/lib/src/keystore/at_key.dart:285]: This sentence is not clear

File: at_commons/lib/src/keystore/at_key.dart [hunk 4]
```diff
@@ -269,25 +276,79 @@ class PrivateKey extends AtKey {
...
   int? ttr;
+
+  /// CCD (Cascade Delete) means if a shared key is deleted, then the corresponding cached key will also be deleted
+  ///
+  /// When set to true, on deleting the original key, the corresponding cached key will be deleted,
+  /// When set to false, the cached key remains even after the original key is deleted.
```
Comment (gkc) [inline at_commons/lib/src/keystore/at_key.dart:291]: Are there any tests which verify this functionality? i.e. is this metadata honoured?

File: at_commons/lib/src/keystore/at_key.dart [hunk 5]
```diff
@@ -269,25 +276,79 @@ class PrivateKey extends AtKey {
...
+  /// This is a derived field representing [ttb] in [DateTime] format
   DateTime? availableAt;
+
+  /// This is a derived field representing [ttl] in [DateTime] format
   DateTime? expiresAt;
+
+  /// This is a derived field representing [ttr] in [DateTime] format
   DateTime? refreshAt;
+
+  /// A date and time representing when the key is created
```
Comment (gkc) [inline at_commons/lib/src/keystore/at_key.dart:303]: `was created` not `is created`

File: at_commons/lib/src/keystore/at_key.dart [hunk 6]
```diff
@@ -269,25 +276,79 @@ class PrivateKey extends AtKey {
...
   DateTime? createdAt;
+
+  /// A date and time representing when the key is last modified
```
Comment (gkc) [inline at_commons/lib/src/keystore/at_key.dart:306]: `was last modified`, not `is last modified`

**Review by gkc — COMMENTED** [review]
I know this PR has already been merged but the documentation that has been added needs to be improved.

**Pull request merged**